### ISSUE TITLE
[Proxy mode] Dualstack endpoints don't get recorded

### ISSUE THREAD

**Issue by aidansteele** [iann0036/iamlive#27]
Hiya Ian,

Truly awesome work on the tool in general and proxy mode specifically. I noticed a couple of issues, shown in the screenshot below.

The first is that the region appears to be determined incorrectly in some cases, e.g. `s3.us-east-1`. The second is that in dualstack mode, the determination of the service definition fails and therefore no parameters get extracted.

![image](https://user-images.githubusercontent.com/369053/113646660-9c04ff00-96cc-11eb-8042-d4322d9437b3.png)

**Comment by iann0036** [comment]
Hey @aidansteele,

Thanks heaps for raising! Looks like I had neglected to consider those path styles. I've updated the codebase to include support that should cover the following URL formats:

* s3.amazonaws.com/bucketname
* bucketname.s3.amazonaws.com
* bucketname.s3-us-east-1.amazonaws.com
* bucketname.s3.us-east-1.amazonaws.com
* bucket.name.s3.us-east-1.amazonaws.com
* s3.us-east-1.amazonaws.com/bucketname
* bucketname.s3.dualstack.us-east-1.amazonaws.com
* s3.dualstack.us-east-1.amazonaws.com/bucketname

Let me know if that fixes the issue for your requests.
